paasta_tools/cli/cmds/spark_run.py

```diff
@@ -711,6 +726,48 @@ def get_spark_app_name(original_docker_cmd: Union[Any, str, List[str]]) -> str:
     return spark_app_name
 
 
+def _convert_memory_string(value: int, input_unit: str, output_unit: str) -> float:
+    units = {
+        "k": 1 << 10,
+        "m": 1 << 20,
+        "g": 1 << 30,
+        "t": 1 << 40,
+    }
+    input_unit = input_unit.lower()
+    output_unit = output_unit.lower()
+    if input_unit not in units.keys():
+        raise ValueError(f"invalid input unit: {input_unit}")
+    if output_unit not in units.keys():
+        raise ValueError(f"invalid output unit: {output_unit}")
+
+    return value * units[input_unit] / units[output_unit]
+
+
+def _calculate_spark_driver_memory_overhead_mb(spark_conf: Mapping[str, str]) -> int:
+    # Reference: https://spark.apache.org/docs/latest/configuration.html
+    memory_overhead_factor = DEFAULT_SPARK_DRIVER_OVERHEAD_FACTOR
+    try:
+        if "spark.driver.memoryOverhead" in spark_conf:
+            return int(
+                max(
+                    MIN_SPARK_DRIVER_MEMORY_OVERHEAD_MB,
+                    spark_conf.get(
+                        "spark.driver.memoryOverhead",
+                        MIN_SPARK_DRIVER_MEMORY_OVERHEAD_MB,
+                    ),
+                )
+            )
+        memory_overhead_factor = float(
+            spark_conf.get("spark.driver.memoryOverheadFactor", memory_overhead_factor)
+        )
+    except Exception as e:
+        log.error(f"error calculating spark driver memory overhead: {e}")
```

Comment:
just fyi: `log.exception` will automatically print out the traceback after the log message :)